### PR TITLE
Add `extract_ast` funnction

### PR DESCRIPTION
**Pull request by otzhora**
This function needed if you want to extract syntactic tree from file

### CODE REVIEW

File: buckwheat/tokenizer.py [hunk 1]
```diff
@@ -266,6 +266,21 @@ def merge_nodes_for_lang(lang: str) -> Set[str]:
             class_types = TreeSitterParser.CLASSES[lang]
         return identifier_types | function_types | class_types
 
+    @staticmethod
+    def get_syntactic_tree_from_file(file: str, lang: str) -> Tuple[tree_sitter.Tree, bytes]:
```
Comment (EgorBu) [inline buckwheat/tokenizer.py]:
```suggestion
    def extract_ast(content: Union[str, bytes], lang: str) -> tree_sitter.Tree:
```
let's separate AST extraction from reading of file.

File: buckwheat/tokenizer.py [hunk 2]
```diff
@@ -266,6 +266,21 @@ def merge_nodes_for_lang(lang: str) -> Set[str]:
...
+        """
+        Given a file and its language, return its syntactic tree and code
```
Comment (EgorBu) [inline buckwheat/tokenizer.py]:
```suggestion
        Given a content and its language, return abstract syntax tree.
```

File: buckwheat/tokenizer.py [hunk 3]
```diff
@@ -266,6 +266,21 @@ def merge_nodes_for_lang(lang: str) -> Set[str]:
...
+        :param file: the path to file.
+        :param lang: the language of code.
+        :return: Tuple[tree_sitter.Tree, bytes]
```
Comment (EgorBu) [inline buckwheat/tokenizer.py]: update it too

**Review by EgorBu — CHANGES_REQUESTED** [review]
Thanks for the PR!) let's adjust it a bit, write test and merge

File: buckwheat/tokenizer.py (hunk 1 above)
Comment (otzhora) [reply buckwheat/tokenizer.py]: If I'm separating ast extraction from reading fille I probably don't need Union[str, bytes]. Or should I read the file if the content is str?

Comment (EgorBu) [reply buckwheat/tokenizer.py]: you should handle it properly - I mean reading and preparation of bytes/str

File: buckwheat/tokenizer.py [hunk 4]
```diff
@@ -266,6 +266,17 @@ def merge_nodes_for_lang(lang: str) -> Set[str]:
             class_types = TreeSitterParser.CLASSES[lang]
         return identifier_types | function_types | class_types
 
+    @staticmethod
+    def extract_ast(content: Union[str, bytes], lang: str) -> tree_sitter.Tree:
+        """
+        Given a content and its language, return its abstract syntactic tree
```
Comment (EgorBu) [inline buckwheat/tokenizer.py:272]:
```suggestion
        Given a content and its language, return its Abstract Syntax Tree
```

**Review by EgorBu — CHANGES_REQUESTED** [review]
Let's fix docstring and merge it!)

**Pull request closed without merging**